WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -262,6 +262,64 @@ def edit
     render :edit
   end
 
+  def payment_setup
+    @competition = Competition.find(params[:id])
+
+    client = create_oauth_client
+    oauth_params = {
+      scope: 'read_write',
+      state: @competition.id,
+    }
+    @authorise_url = client.auth_code.authorize_url(oauth_params)
+  end
+
+  def revoke_stripe_access
+    @competition = Competition.find(params[:id])
+    if @competition.connected_stripe_account_id
+      Stripe.api_key = ENVied.STRIPE_API_KEY
+      account = Stripe::Account.retrieve(@competition.connected_stripe_account_id)
+      account.deauthorize(
+        ENVied.STRIPE_CLIENT_ID,
+        { stripe_user_id: @competition.connected_stripe_account_id},
+      )
+      @competition.connected_stripe_account_id = nil
+      @competition.save
+    end
+    redirect_to competitions_payment_setup_path
+  end
+
+  def stripe_connect
+    code = params[:code]
+    competition = Competition.find(params[:state])
+    if !competition.user_can_view?(current_user)
+      raise ActionController::RoutingError.new('Not Found')
+    end
+    client = create_oauth_client
+    resp = client.auth_code.get_token(code, params: {scope: 'read_write'})
+    competition.connected_stripe_account_id = resp.params['stripe_user_id']
+    if competition.save
+      flash[:success] = t('competitions.messages.stripe_connected')
+    else
+      flash[:success] = t('competitions.messages.stripe_not_connected')
+    end
+    redirect_to competitions_payment_setup_path(competition)
+  end
+
+  private def create_oauth_client
```

Comment:
Nit: since the WCA website does have a decent amount of code about WCA OAuth, I think it might be good to rename this method to make it clear that it's for stripe oath.